bumble/device.py

```diff
@@ -2230,45 +2230,43 @@ def on_connection(
             self.advertising_own_address_type = None
             self.advertising = False
 
-            # Create and notify of the new connection asynchronously
-            async def new_connection():
+            if own_address_type in (
+                OwnAddressType.PUBLIC,
+                OwnAddressType.RESOLVABLE_OR_PUBLIC,
+            ):
+                self_address = self.public_address
+            else:
+                self_address = self.random_address
+
+            # Create a new connection
+            connection = Connection(
+                self,
+                connection_handle,
+                transport,
+                self_address,
+                peer_address,
+                peer_resolvable_address,
+                role,
+                connection_parameters,
+                ConnectionPHY(HCI_LE_1M_PHY, HCI_LE_1M_PHY),
+            )
+            self.connections[connection_handle] = connection
+
+            # Emit an event to notify listeners of the new connection
+            self.emit('connection', connection)
```

Comment:
the problem here is that we're going to emit an initial connection with possibly the wrong PHY, then update its PHY, but the event has already been emitted, so the listener won't seen the change. Even if we emitted an update, it would look like a PHY transition, which isn't really right, since the initial PHY may not be 1M.

Reply:
I can delay the emit after the read if supported, and just emit otherwise ? That was my first version

Reply:
Done :+1: